features/filesystem/bd/ObservingBlockDevice.cpp

```diff
@@ -43,11 +43,19 @@ void ObservingBlockDevice::attach(Callback<void(BlockDevice *)> cb)
 
 int ObservingBlockDevice::init()
 {
+    if (_init_ref_count++) {
+        return BD_ERROR_OK;
+    }
+
```

Comment:
@davidsaada, I think I'm missing something. Is this needed if we can expect the underlying block device to also be reference counted?

Reply:
Yes it is. If the block device has a resource (like `BufferedBlockDevice`) it would make sense to protect it against the scenario described in the issue.
Now, while it is not needed in devices like `ObservingBlockDevice`, ones not having any allocated resources, it is harmless, and it will become effective if one decides to add such a resource in the future. This is why I think it's a good practice to have this in all block devices, also for reference sake for future ones (which may need it).